### PR TITLE
fixed the missing fields in GeneralInfo component fixed #370

### LINKED ISSUE #370: Record when missing general info fields
When no DOI on a record page: either hide the field of show No DOI;
When no abbreviation, remove the empty parenthesis.
same for all the other fields: display either None or hide the field if no data is available.
@terazus would be good to have a picture of the route query to see what record had the problem

### PR DESCRIPTION
**Pull request by hoseinmirian**

### CODE REVIEW

File: src/components/Records/Record/GeneralInfo.vue [hunk 1]
```diff
@@ -38,16 +38,30 @@
         <div class="d-flex flex-column mt-2  ml-sm-6 ml-lg-8">
           <div class="d-flex flex-row mb-2 align-center">
             <h3>{{ currentRecord['fairsharingRecord'].name }}</h3>
-            <b class="ml-2">({{ currentRecord['fairsharingRecord'].abbreviation
-            }})</b>
+            <h3
```
Comment (terazus) [inline src/components/Records/Record/GeneralInfo.vue]: You can remove this line as name is a required field (there's always a name)

**Review by terazus — CHANGES_REQUESTED** [review]
Missing others fields in general info:
- year of creation
- countries
- hompeage
(registry and type are mandatory and should always be there).
Maybe refactor these elements into a component that take a field name and field value as a input proeprties and display the corresponding line.

File: src/components/Records/Record/GeneralInfo.vue (hunk 1 above)
Comment (hoseinmirian) [reply src/components/Records/Record/GeneralInfo.vue]: image
done

File: src/components/Records/Record/GeneralInfo.vue [hunk 2]
```diff
@@ -38,16 +38,23 @@
         <div class="d-flex flex-column mt-2  ml-sm-6 ml-lg-8">
           <div class="d-flex flex-row mb-2 align-center">
             <h3>{{ currentRecord['fairsharingRecord'].name }}</h3>
-            <b class="ml-2">({{ currentRecord['fairsharingRecord'].abbreviation
-            }})</b>
+            <b class="ml-2">({{ currentRecord['fairsharingRecord'].abbreviation }})</b>
+            <NoneFound :string-field="currentRecord['fairsharingRecord'].abbreviation" />
+            <span
```
Comment (terazus) [inline src/components/Records/Record/GeneralInfo.vue]: This line isn't needed anymore

File: src/components/Records/Record/GeneralInfo.vue [hunk 3]
```diff
@@ -38,16 +38,23 @@
         <div class="d-flex flex-column mt-2  ml-sm-6 ml-lg-8">
           <div class="d-flex flex-row mb-2 align-center">
             <h3>{{ currentRecord['fairsharingRecord'].name }}</h3>
-            <b class="ml-2">({{ currentRecord['fairsharingRecord'].abbreviation
-            }})</b>
+            <b class="ml-2">({{ currentRecord['fairsharingRecord'].abbreviation }})</b>
```
Comment (terazus) [inline src/components/Records/Record/GeneralInfo.vue]: Should `<b v-if='val'> <NoneFound v-else>` to avoid empty divs

Comment (terazus) [reply src/components/Records/Record/GeneralInfo.vue]: Also remove NoneFound for abbreviation.

**Review by terazus — CHANGES_REQUESTED** [review]

**Review by knirirr — CHANGES_REQUESTED** [review]
On my machine it seems as if the description field is not aligned with the label. Here's an example:

Screenshot 2020-08-18 at 16 24 33

**Comment by hoseinmirian** [comment]
> 
> 
> On my machine it seems as if the description field is not aligned with the label. Here's an example:
> 
> Screenshot 2020-08-18 at 16 24 33

record number please.

**Comment by hoseinmirian** [comment]
@knirirr the bug was due to p tag default margin and now is fixed

**Review by knirirr — APPROVED** [review]

File: src/components/Records/Record/GeneralInfo.vue [hunk 4]
```diff
@@ -38,16 +38,25 @@
...
+            >({{ currentRecord['fairsharingRecord'].abbreviation }})</b>
           </div>
           <div class="d-flex align-center">
             <h3 class="mr-1">
               doi:
             </h3>
-            <a :href="currentRecord['fairsharingRecord'].doi">
+            <a
+              v-if="currentRecord['fairsharingRecord'].doi"
+              :href="currentRecord['fairsharingRecord'].doi"
```
Comment (terazus) [inline src/components/Records/Record/GeneralInfo.vue:52]: this need to have target="_blank" as the DOI will very likely be an external link

File: src/components/Records/Record/GeneralInfo.vue [hunk 5]
```diff
@@ -66,29 +75,50 @@
...
           target="_blank"
         >{{ currentRecord['fairsharingRecord'].homepage }}</a>
+        <NoneFound
+          v-else
+          :string-field="currentRecord['fairsharingRecord'].homepage"
+        />
       </div>
       <!--Developed Countries -->
       <div class="d-flex flex-wrap">
         <b class="mr-2">Countries involved with this resource:</b>
```
Comment (terazus) [inline src/components/Records/Record/GeneralInfo.vue:112]: Bug with countries but i think it's be issued already:
image
for record 1234.

File: tests/unit/components/Records/Record/GeneralInfo.spec.js [hunk 6]
```diff
@@ -29,8 +29,9 @@ describe("GeneralInfo.vue", function(){
 
     it("can be instantiated", () => {
         expect(wrapper.name()).toMatch("GeneralInfo");
+        console.log(wrapper.vm.currentRecord['fairsharingRecord'].metadata.year_creation)
```
Comment (terazus) [inline tests/unit/components/Records/Record/GeneralInfo.spec.js]: Please no console.log statements :)

File: tests/unit/components/Records/Record/NoneFound.spec.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
-import { shallowMount } from "@vue/test-utils";
+import {shallowMount} from "@vue/test-utils";
```
Comment (terazus) [inline tests/unit/components/Records/Record/NoneFound.spec.js]: please do { shallowMount } with the correct whitespacing.

**Review by terazus — CHANGES_REQUESTED** [review]
Please review  tests/unit/components/Records/Record/NoneFound.spec.js as this test suite is very messy.

**Review by terazus — CHANGES_REQUESTED** [review]
tests/unit/components/Records/Record/NoneFound.spec.js  is still very messy.
There's two beforeEach declarations. There's two instanciations tests with different wrappers but the same description. There's two tests without any logic.

File: src/components/Records/Record/NoneFound.vue [hunk 8]
```diff
@@ -1,35 +1,55 @@
...
-        }
-      }
+export default {
+  name: "NoneFound",
+  props: {
+    dataField: {
+      type: Array,
+      default: null
+    },
+    objectField: {
```
Comment (terazus) [inline src/components/Records/Record/NoneFound.vue]: what is this and why is it not used anywhere?

**Review by terazus — CHANGES_REQUESTED** [review]
There's an issue for newly created records where nothing appears:
image
compared to dev:
image
(either create a new record or use api.fairsharing.org with record id 3166)

File: src/components/Records/Record/NoneFound.vue (hunk 8 above)
Comment (hoseinmirian) [reply src/components/Records/Record/NoneFound.vue]: yes this got to be removed.

**Review by terazus — CHANGES_REQUESTED** [review]
Behaviour still present, test with ID = 3166 on remote the remote api (not the vm).

**Pull request merged**